File: Android Java/Testing/app/src/main/java/com/example/testing/picker/DatePickerFragment.java
package com.example.testing.picker;

import android.app.DatePickerDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.widget.DatePicker;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.FragmentManager;

import java.util.Calendar;

public class DatePickerFragment extends DialogFragment {
    private DatePickerDialog.OnDateSetListener listener;

    public static DatePickerFragment newInstance(DatePickerDialog.OnDateSetListener listener) {
        DatePickerFragment fragment = new DatePickerFragment();
        fragment.setListener(listener);
        return fragment;
    }

    public void setListener(DatePickerDialog.OnDateSetListener listener) {
        this.listener = listener;
    }

    @Override
    @NonNull
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        final Calendar c = Calendar.getInstance();
        int year = c.get(Calendar.YEAR);
        int month = c.get(Calendar.MONTH);
        int day = c.get(Calendar.DAY_OF_MONTH);

        return new DatePickerDialog(getActivity(), listener, year, month, day);
    }
    public static void showDatePickerDialog(FragmentManager activity, final TextView tv) {
        final String[] selectedDate = {""};
        DatePickerFragment newFragment = DatePickerFragment.newInstance(new DatePickerDialog.OnDateSetListener() {
            @Override
            public void onDateSet(DatePicker datePicker, int year, int month, int day) {
                String me="";
                me=String.valueOf(month+1);
                if((month+1)<10)
                {
                    me="0"+(month+1);
                }
                selectedDate[0] = year + "-" + (me) + "-" + day;
                tv.setText(selectedDate[0]);
            }
        });
        newFragment.show(activity, "datePicker");
    }
}

File: Android Kotlin/Testing - copia/app/src/main/java/com/example/testing/picker/DatePickerFragment.kt
package com.example.testing.picker

import android.app.DatePickerDialog
import android.app.DatePickerDialog.OnDateSetListener
import android.app.Dialog
import android.os.Bundle
import android.widget.TextView
import androidx.fragment.app.DialogFragment
import androidx.fragment.app.FragmentManager
import java.util.*

class DatePickerFragment : DialogFragment() {
    private var listener: OnDateSetListener? = null
    fun setListener(listener: OnDateSetListener?) {
        this.listener = listener
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val c = Calendar.getInstance()
        val year = c[Calendar.YEAR]
        val month = c[Calendar.MONTH]
        val day = c[Calendar.DAY_OF_MONTH]
        return DatePickerDialog(activity!!, listener, year, month, day)
    }

    companion object {
        fun newInstance(listener: OnDateSetListener?): DatePickerFragment {
            val fragment = DatePickerFragment()
            fragment.setListener(listener)
            return fragment
        }

       // @kotlin.jvm.JvmStatic
        fun showDatePickerDialog(activity: FragmentManager?, tv: TextView) {
            val selectedDate = arrayOf("")
            val newFragment = newInstance { datePicker, year, month, day ->
                var me = ""
                me = (month + 1).toString()
                if (month + 1 < 10) {
                    me = "0" + (month + 1)
                }
                selectedDate[0] = "$year-$me-$day"
                tv.text = selectedDate[0]
            }
            newFragment.show(activity!!, "datePicker")
        }
    }
}